lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -299,6 +305,8 @@ private int nextBatch()
     private boolean advanceToNextRowGroup()
             throws IOException
     {
+        currentRowGroupMemoryContext.close();
+        currentRowGroupMemoryContext = memoryContext.newAggregatedMemoryContext();
```

Comment:
don't you have to release column readers: `columnReaders.clear();`?

Reply:
`initializeColumnReaders` overwrites `columnReaders` in this code path with new column readers, that releases references to the previous column readers